src/OpenTK.Windowing.Common/Monitor.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace OpenToolkit.Windowing.Common
+{
+    /// <summary>
+    /// Wrapper around an implementation-defined monitor struct.
+    /// </summary>
+    public struct Monitor
+    {
+        public Monitor(IntPtr ptr)
+        {
+            Ptr = ptr;
```

Comment:
Was that a decided change, or why not use Monitor* ptr anymore like it was done before?

Imho it is now more work to use this wrapper struct, loses a little bit of typesafety(takes any IntPtr), and performance wise it should be a little bit worse as well.

I personally liked, how it was done before.

Reply:
We discussed it in the Discord and figured this was the best way forward.

Reply:
We need some way of setting Monitor in the constructor, so it has to be part of whatever struct we use to create the NativeWindow. We can't include a GLFWmonitor* because GLFW can't be used in Windowing.Common.

Reply:
Ohh right forgot about that. But I'm thinking as well that perhaps an Interface IMonitor would be better while Windowing.Desktop uses a Monitor struct like that(and could obviously then use normal pointers again). Because IntPtr is not really abstract and probably not valid for all systems. Ofc e.g. Android has no Monitors anyway, but in wayland a FileDescriptor can be used to reference to a Monitor, while we not necessarily want to write our own Wayland implementation, in my book the abstraction argument still stands and in my opinion is worth a little bit of discussion still ;)

Reply:
I like your suggestion more, I'll implement this.

Reply:
Attempted to implement this, but it ended up not working.

If I'm understand you correctly, this is what it would involve: IMonitor would need to be empty - no methods or properties at all. The current Monitor struct would then be moved over to Windowing.Desktop, unchanged except for implementing IMonitor and directly using a GLFWwindow* instead of an IntPtr. IWindowProperties would then use IMonitor instead.

This approach didn't work. Because IMonitor has no properties, I can't end up get the pointer on Monitor.

Theoretically, I could check if the passed IMonitor is a Monitor, and if it is, cast to that and get the pointer that way. However, I'm concerned about potential performance issues, and it might be confusing for only a single derived class to be usable - it sort of goes against the concept of an interface.

If there's another way of implementing the IMonitor approach you mentioned, I would be glad to hear it.

Reply:
True I forgot about that Problem. I currently can't think of a solution to this in the current C# language. Possibly in a later version(as necessary features are already in discussion - e.g. Shapes or Concepts)